src/net/spdy/spdy_session.cc

```diff
@@ -2172,11 +2184,30 @@ void SpdySession::EnqueueResetStreamFrame(spdy::SpdyStreamId stream_id,
   });
 
   DCHECK(buffered_spdy_framer_.get());
+  int padding_len = base::RandInt(48, 72) - spdy::kDataFrameMinimumSize;
   std::unique_ptr<spdy::SpdySerializedFrame> rst_frame(
-      buffered_spdy_framer_->CreateRstStream(stream_id, error_code));
+      buffered_spdy_framer_->CreateRstStream(stream_id, error_code,
+                                             padding_len));
+
+  auto buffer = std::make_unique<SpdyBuffer>(std::move(rst_frame));
+
+  DecreaseSendWindowSize(padding_len);
```

Comment:
This is subtly wrong. RFC 7540 § 6.1 says: "The entire DATA frame payload is included in flow control, including the Pad Length and Padding fields if present."

You did not include the Pad Length field, which is an off-by-one error.

Reply:
Just so you know this is not asking you to correct your patch. I'll have to reimplement the same design.

Reply:
> off-by-one error

It seems like, but isn't. See:

https://github.com/klzgrad/naiveproxy/blob/cc7da21ff6492daece4e803f540e217cc793d7e1/src/net/third_party/quiche/src/spdy/core/spdy_protocol.h#L543-L549